File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        count = {}
        for ind, val in enumerate(nums):
            if target - val not in count:
                count[val] = ind
            else:
                return [count[target-val], ind]
    
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const prevMap = new Map();
    for (let i = 0; i < nums.length; i ++){
        let diff = target - nums[i]
        if (prevMap.has(diff)){
            return [prevMap.get(diff), i];
        }
        else{
            prevMap.set(nums[i], i);
        }
    }
    
};